model/flow/identity.go

```diff
@@ -591,66 +541,20 @@ func (il IdentityList) Exists(target *Identity) bool {
 // target:  value to search for
 // CAUTION:  The identity list MUST be sorted prior to calling this method
 func (il IdentityList) IdentifierExists(target Identifier) bool {
-	left := 0
-	lenList := len(il)
-	if lenList == 0 {
-		return false
-	}
-
-	right := lenList - 1
-	mid := right >> 1
-	num2 := target[:]
-
-	// pre-calculate these 4 values for comparisons later
-	var tgt [4]uint64
-	tgt[0] = binary.BigEndian.Uint64(num2[:])
-	tgt[1] = binary.BigEndian.Uint64(num2[8:])
-	tgt[2] = binary.BigEndian.Uint64(num2[16:])
-	tgt[3] = binary.BigEndian.Uint64(num2[24:])
-
-	for {
-		num1 := il[mid].NodeID[:]
-		lenID := len(num1)
-		i := 0
-
-		for {
-			chunk1 := binary.BigEndian.Uint64(num1[i:])
-			chunk2 := tgt[i/8]
-
-			if chunk1 < chunk2 {
-				left = mid + 1
-				break
-			} else if chunk1 > chunk2 {
-				right = mid - 1
-				break
-			} else if i >= lenID-8 {
-				// we're on the last chunk of 8 bytes, and
-				// so return true if equal -- it exists
-				return true
-			}
-
-			// these 8 bytes were equal, so increment index by 8 bytes
-			i += 8
-		}
-		if left > right {
-			return false
-		}
-		mid = (left + right) >> 1
-	}
+	_, ok := slices.BinarySearchFunc(il, &Identity{NodeID: target}, func(a, b *Identity) int {
+		return bytes.Compare(a.NodeID[:], b.NodeID[:])
+	})
+	return ok
 }
 
 // GetIndex returns the index of the identifier in the IdentityList and true
 // if the identifier is found.
-func (il IdentityList) GetIndex(identifier Identifier) (uint, bool) {
-	index := 0
-	ok := false
-	for i, id := range il.NodeIDs() {
-		if id == identifier {
-			index = i
-			ok = true
-			break
-		}
+func (il IdentityList) GetIndex(target Identifier) (uint, bool) {
+	i := slices.IndexFunc(il, func(a *Identity) bool {
+		return bytes.Equal(a.NodeID[:], target[:])
```

Comment:
`Identifier` is a fixed-length array, which can be directly compared in go without the need to convert to slices:
```suggestion
		return a.NodeID == target
```